doc/feature/ReqRep.md

```diff
@@ -0,0 +1,112 @@
+
+    class Client : public Receiver
+    {
+        // ctors same as Subscriber
+        void get(const Serializable& request, ReplyFunc& func);
+    };
+
+    using HandleFunc = std::function<servus::Serializable::Data(const void*, size_t)>;
+
+    class Server : public Receiver
+    {
+        // ctors as Publisher
+        void handle(const uint128_t& request, HandleFunc& func)
+    };
+
+
+## Implementation
+
+* Maps closely to ZeroMQ
+  * Client uses a dealer socket, receive() will call ReplyFunc
+  * Server::receive() uses a rep socket
+    * Single-threaded, synchronous for now
+    * Can be extended to async model later
+    * does recv()-send()
+* Client::get is fully asynchronous
+* Client::get will provide nullptr, 0 if server has no handle func or handle func returns no data
+
+## Examples
+
+    class livre::RemoteDataSource
+    {
+        RemoteDataSource(const DataSourcePluginData& initData)
+        {
+            if( !_client.get( initData.uri, initData.info ))
+                throw std::runtime_error( "Open failed");
+        }
+
+        MemoryUnitPtr getData(const LODNode& node)
+        {
+            MemoryUnitPtr block;
+            _client.get({node, _uri},
+                        [&]( const void* data, const size_t size ){
+                            block = new AllocMemoryUnit(size);
+                            memcpy data to block
+                        });
+
+            _client.receive( TIMEOUT );
+            return block;
+        }
+
+        zeroeq::Client _client;
+    };
+
+    class livre::RemoteDataService
+    {
+        RemoteDataService(const URI& uri) // does not return
+        {
+            _server.handle(URI::getTypeIdentifier(), &_getInfo);
+            _server.handle(LODNode::getTypeIdentifier(), &_getData);
+
+            while( _server.receive()) /*nop*/ ;
+        }
+
+        zeroeq::Server _server;
+
+        servus::Serializable::Data _getInfo(const void* data, const size_t size)
+        {
+            ConstURIPtr uri = livre::URI::create(data, size);
+            if( openAndCacheDatasource( *uri ))
+                return dataSource.getVolumeInfo().toBinary();
+            return {};
+        }
+
+        servus::Serializable::Data _getData(const void* data, const size_t size)
+        {
+            ConstLODNodeURIPtr node = livre::LODNodeURI::create(data, size);
+            // open data source if needed, error handling, ...
+            return dataSource.getData(*node).toBinary();
+        }
+    };
+
+## Issues
+
+### 1: Shall the handle function return a future like the http::Server?
+
+_Resolution: No:_ The http::Server manages internal thread(s) for the protocol
+handling. The zmq rep socket does not, and a multithreaded server would still call handle() synchronously from each worker thread.
+
+### 2: Is the client a receiver?
+
+_Resolution: Yes:_ The get method only sends the request, the reply is received in receive() and correlated to the request.
```

Comment:
If a client issues millions of requests and doesn't call receive at all, what does it happen on the server?